packages/control-group/src/styles.tsx

```diff
@@ -0,0 +1,274 @@
+    case 'radio-horizontal':
+    case 'checkbox-horizontal':
+      return getHorizontalGroupStyle(_)
+    case 'radio-vertical':
+    case 'checkbox-vertical':
+      return getVerticalGroupStyle(_)
+    default:
+      return {}
+  }
+}
+
+export const ControlGroupWrapper = styled.div<WrapperProps>`
+  ${_ => ({
+    ...getControlGroupStyle(_)
+  })}
+`
+
+const getBorder = (_: ThemeProps & ButtonProps) => {
+  const {outline} = _.theme.colors.button
+  const border = (state: keyof typeof outline.borderColor) =>
+    `${outline.borderWidth}px solid ${outline.borderColor[state]}`
+
+  const state = getState(_) as keyof typeof outline.borderColor
+  let borderStyle = border(state)
+
+  return {
+    borderTop: borderStyle,
+    borderBottom: _.isOpen ? 'none' : borderStyle,
+    borderRight: _.isOpen ? borderStyle : border(state),
+    borderLeft: _.isOpen ? borderStyle : 'none',
+
+    '&:first-of-type': {
+      borderLeft: borderStyle
+    },
+    '&:nth-last-of-type(2)': {
+      borderBottom: borderStyle
+    },
+    '&:last-of-type': {
+      borderRight: _.isOpen ? 'none' : borderStyle
+    }
+  }
+}
+
+const getBorderRadius = (_: ThemeProps & {isOpen?: boolean}) => ({
+  borderRadius: 0,
+
+  '&:first-of-type': {
+    borderRadius: _.isOpen
+      ? `${_.theme.radius} ${_.theme.radius} 0 0`
+      : `${_.theme.radius} 0 0 ${_.theme.radius}`
+  },
+
+  '&:nth-last-of-type(2)': {
+    borderRadius: _.isOpen ? `0 0 ${_.theme.radius} ${_.theme.radius}` : 0
+  },
+
+  '&:last-of-type': {
+    borderRadius: `0 ${_.theme.radius} ${_.theme.radius} 0`,
+    ...(_.isOpen
+      ? {
+          '&, &:hover, &:active, &[aria-invalid="true"]': {
+            border: 0,
+            borderRadius: 0
+          }
+        }
+      : {})
+  }
+})
+
+const Button = styled(SmashingButton)<ButtonProps>`
+  box-shadow: none;
+  position: relative;
+  white-space: nowrap;
+
+  ${_ => {
+    let style
+    switch (_.appearance) {
+      case 'outline':
+        style = {
+          boxSizing: 'border-box',
+          ...getBorderRadius(_),
+
+          '&, &:hover, &:active, &[aria-invalid="true"], &:disabled': {
+            boxShadow: 'none',
+            ...getBorder(_)
+          }
+        }
+        break
+      default:
+        style = {
+          ...getBorderRadius(_),
+          ...(_.checked ? getButtonStyle(_.appearance)(_)[':active'] : {})
+        }
+    }
+
+    return style
+  }}
+
+  &:focus {
+    box-shadow: none;
+  }
+`
+
+export const ControlButton = styled(Button)<ButtonProps>`
+  justify-content: ${_ => getTextAlign(_.textAlign)};
+
+  ${_ => {
+    const activeLine = () => ({
+      color: getTextColor(_.invalid, _.disabled)(_),
+
+      '&:after': {
+        position: 'absolute',
+        content: '" "',
+        bottom: 0,
+        left: _.theme.spacing.xxs,
+        width: `calc(100% - ${_.theme.spacing.sm})`,
+        height: '2px',
+        backgroundColor: 'currentColor'
+      }
+    })
+
+    let style
```

Comment:
Just use `return` in each case - no need to assign it to variable

Reply:
fixed